File: src/main/java/net/jlefever/dsmutils/depends/Dep.java
package net.jlefever.dsmutils.depends;

import java.util.Objects;

public class Dep {
    private final Entity source;
    private final Entity target;
    private final String kind;

    public Dep(Entity source, Entity target, String kind)
    {
        this.source = source;
        this.target = target;
        this.kind = kind;
    }

    public Entity getSource()
    {
        return source;
    }

    public Entity getTarget()
    {
        return target;
    }

    public String getKind()
    {
        return kind;
    }

    @Override
    public boolean equals(Object obj)
    {
        return Objects.hashCode(this) == Objects.hashCode(obj);
    }

    @Override
    public int hashCode()
    {
        return Objects.hash(getSource(), getTarget(), getKind());
    }

    @Override
    public String toString()
    {
        return getSource().toString() + " <" + getKind() + "> " + getTarget().toString();
    }
}

File: frontend2/src/models/Dep.ts
import Entity from "./Entity";

export default class Dep {
    private readonly _id: number;
    private readonly _source: Entity;
    private readonly _target: Entity;
    private readonly _kind: string;
    private readonly _weight: number;

    constructor(id: number, source: Entity, target: Entity, kind: string, weight: number) {
        this._id = id;
        this._source = source;
        this._target = target;
        this._kind = kind;
        this._weight = weight;
    }

    public get id() {
        return this._id;
    }

    public get source() {
        return this._source;
    }

    public get target() {
        return this._target;
    }

    public get kind() {
        return this._kind;
    }

    public get weight() {
        return this._weight;
    }
}